packages/vite/src/node/config.ts

```diff
@@ -703,6 +691,22 @@ function mergeConfigRecursively(
       }
     }
 
+    if (Array.isArray(existing) || Array.isArray(value)) {
+      merged[key] = [
+        ...(Array.isArray(existing) ? existing : [existing]),
+        ...(Array.isArray(value) ? value : [value])
```

Comment:
I think we can use the `arraify` utility to simplify this a bit more:

```suggestion
        ...arraify(existing),
        ...arraify(value)
```

Reply:
done